core/src/main/scripts/migrate_db.py

```diff
@@ -0,0 +1,204 @@
+        print >> ERROR_FILE, msg
+        return None
+
+    if connection is not None:
+        return connection.cursor()
+
+def get_portal_properties(properties_filename):
+    """ Returns a properties object """
+    
+    properties = {}
+    properties_file = open(properties_filename, 'r')
+
+    for line in properties_file:
+        line = line.strip()
+
+        # skip line if its blank or a comment
+        if len(line) == 0 or line.startswith('#'):
+            continue
+        
+        # store name/value
+        property = line.split('=')
+        if len(property) != 2:
+            print >> ERROR_FILE, 'Skipping invalid entry in proeprty file: ' + line
+            continue
+        properties[property[0]] = property[1].strip()
+    properties_file.close()
+
+    if (DATABASE_HOST not in properties or len(properties[DATABASE_HOST]) == 0 or
+        DATABASE_NAME not in properties or len(properties[DATABASE_NAME]) == 0 or
+        DATABASE_USER not in properties or len(properties[DATABASE_USER]) == 0 or
+        DATABASE_PW not in properties or len(properties[DATABASE_PW]) == 0):
+        print >> ERROR_FILE, 'Missing one or more required properties, please check property file'
+        return none
+    
+    # return an instance of PortalProperties
+    return PortalProperties(properties[DATABASE_HOST],
+                            properties[DATABASE_NAME],
+                            properties[DATABASE_USER],
+                            properties[DATABASE_PW])
+
+def get_db_version(cursor):
+    """ gets the version number of the database """
+
+    # First, see if the version table exists
+    version_table_exists = False
+    try:
+        cursor.execute('select table_name from information_schema.tables')
+        for row in cursor.fetchall():
+            if VERSION_TABLE == row[0].lower().strip():
+                version_table_exists = True
+    except MySQLdb.Error, msg:
+        print >> ERROR_FILE, msg
+        return None
+    
+    if not version_table_exists:
+        return (0,0,0)
+
+    # Now query the table for the version number
+    try:
+        cursor.execute('select ' + VERSION_FIELD + ' from ' + VERSION_TABLE)
+        for row in cursor.fetchall():
+            version = tuple(map(int, row[0].strip().split('.')))
+    except MySQLdb.Error, msg:
+        print >> ERROR_FILE, msg
+        return None
+
+    return version
+
+def is_version_larger(version1, version2):
+    """ Checks if version 1 is larger than version 2 """
+    
+    if version1[0] > version2[0]:
+        return True
+    if version2[0] > version1[0]:
+        return False
+    if version1[1] > version2[1]:
+        return True
+    if version2[1] > version1[1]:
+        return False
+    if version1[2] > version2[2]:
+        return True
+    return False
+
+def run_migration(db_version, sql_filename, cursor):
+    """
+
+        Goes through the sql and runs lines based on the version numbers. SQL version should be stated as follows:
```

Comment:
Database schema change should be very rare and any change will be reviewed. I'd not worry about someone forgetting to add the version line.

You can add multiple lines following the version line.